sql/hive/src/test/scala/org/apache/spark/sql/hive/ExpressionToSQLSuite.scala

```diff
@@ -268,4 +268,9 @@ class ExpressionToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     checkSqlGeneration("SELECT input_file_name()")
     checkSqlGeneration("SELECT monotonically_increasing_id()")
   }
+
+  test("subquery") {
+    checkSqlGeneration("SELECT 1 + (SELECT 2)")
+    checkSqlGeneration("SELECT 1 + (SELECT 2 + (SELECT 3 as a))")
```

Comment:
Could we add more test cases?

```SELECT * from (SELECT 1 + (SELECT 2 + (SELECT 3 as a)) t``` 

If we run this, we hit a bug in Scalar Subquery support. If you want, I can try to fix it.